src/server/mysql_conn.cpp

```diff
@@ -0,0 +1,469 @@
+/*
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#ifdef HAVE_MYSQLPP
+
+#include "mysql_conn.hpp"
+
+#include "log.hpp"
+
+static lg::log_domain log_sql_handler("sql_executor");
+#define ERR_SQL LOG_STREAM(err, log_sql_handler)
+#define WRN_SQL LOG_STREAM(warn, log_sql_handler)
+#define LOG_SQL LOG_STREAM(info, log_sql_handler)
+#define DBG_SQL LOG_STREAM(debug, log_sql_handler)
+
+mysql_conn::mysql_conn(const config& c)
+	: db_users_table_(c["db_users_table"].str())
+	, db_banlist_table_(c["db_banlist_table"].str())
+	, db_extra_table_(c["db_extra_table"].str())
+	, db_game_info_table_(c["db_game_info_table"].str())
+	, db_game_player_info_table_(c["db_game_player_info_table"].str())
+	, db_game_modification_info_table_(c["db_game_modification_info_table"].str())
+	, db_user_group_table_(c["db_user_group_table"].str())
+	, db_tournament_query_(c["db_tournament_query"].str())
+	, mp_mod_group_(0)
+{
+	try
+	{
+		// NOTE: settings put on the connection are NOT kept if a reconnect occurs!
```

Comment:
Is there a workaround for this? Otherwise the encoding will be an issue, no?

Reply:
I'll clarify the comment, but what it means is that while settings put on the connection aren't kept on reconnect, settings put on the account are kept.  That's why it uses the account's `set_connect_option` and `set_schema` methods instead of the connection's `set_charset` and `set_schema` methods, respectively.